File: hibernate-scala-examples/src/main/scala/org/hibernate/examples/hibernate/HibernateParameter.scala
package org.hibernate.examples.hibernate

import org.hibernate.`type`.StandardBasicTypes
import org.hibernate.examples.AbstractNamedParameter

/**
 * [[org.hibernate.Query]]에 사용할 Hibernate용 Parameter 정보를 표현합니다.
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since  2013. 12. 19. 오후 9:39
 */
@SerialVersionUID(3975859046850492000L)
class HibernateParameter(override val name: String,
                         override val value: Any,
                         val paramType: org.hibernate.`type`.Type = StandardBasicTypes.SERIALIZABLE)
    extends AbstractNamedParameter(name, value) {

    def this(name: String, value: Any) {
        this(name, value, StandardBasicTypes.SERIALIZABLE)
    }

    override protected def buildStringHelper =
        super.buildStringHelper.add("paramType", paramType)
}

File: hibernate-examples/src/main/java/org/hibernate/examples/hibernate/HibernateParameter.java
package org.hibernate.examples.hibernate;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.examples.AbstractNamedParameter;
import org.hibernate.examples.utils.ToStringHelper;
import org.hibernate.type.StandardBasicTypes;

/**
 * Hibernate Query 에 사용할 인자 정보를 표현합니다.
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 2013. 11. 28. 오후 1:59
 */
public class HibernateParameter extends AbstractNamedParameter {

  @Getter
  @Setter
  private org.hibernate.type.Type paramType = StandardBasicTypes.SERIALIZABLE;

  public HibernateParameter(String name, Object value) {
    super(name, value);
  }

  public HibernateParameter(String name, Object value, org.hibernate.type.Type paramType) {
    super(name, value);
    this.paramType = paramType;
  }

  @Override
  public ToStringHelper buildStringHelper() {
    return super.buildStringHelper().add("paramType", paramType);
  }

  private static final long serialVersionUID = -228167603287695079L;
}
